pkg/render/manager.go

```diff
@@ -352,6 +355,21 @@ func (c *managerComponent) managerDeployment() *appsv1.Deployment {
 
 	if c.cfg.Manager != nil {
 		if overrides := c.cfg.Manager.Spec.ManagerDeployment; overrides != nil {
+			// Due to the change of prefix for some components (eg. tigera-manager -> calico-manager) we need to support
+			// the upgrade scenario in a graceful way that doesn't force the user to update the new names. To accomplish
+			// this we will simply perform the substitution from old name to new name before applying the settings to
+			// the deployment
+
+			if overrides.Spec != nil && overrides.Spec.Template != nil && overrides.Spec.Template.Spec != nil {
```

Comment:
Clever.

It looks like we never migrated es-proxy to ui-apis here. I think it is better to add a substitution for that as well.